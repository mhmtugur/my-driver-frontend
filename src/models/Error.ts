export class Error {
  errorConstant: string
  errorMessage: string

  constructor({ errorConstant, errorMessage }){
    this.errorConstant = errorConstant;
    this.errorMessage = errorMessage;
  }
}