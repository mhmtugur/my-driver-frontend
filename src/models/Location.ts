export class Location {
  id: number
  originPlaceId: string
  city: string
  category: string
  label: string
  address: string

  constructor({ id, originPlaceId, city, category, label, address}) {
    this.id = id;
    this.originPlaceId = originPlaceId;
    this.city = city;
    this.category = category;
    this.label = label;
    this.address = address;
  }
}