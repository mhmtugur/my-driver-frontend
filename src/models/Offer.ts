export class Offer {
  amount: number
  currency: string
  title: string
  exampleCar : string
  web_img: string

  constructor({ amount, currency, title, exampleCar, web_img }) {
    this.amount = amount
    this.currency = currency;
    this.title = title;
    this.exampleCar = exampleCar;
    this.web_img = web_img;
  }
}