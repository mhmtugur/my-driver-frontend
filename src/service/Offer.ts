import axios from 'axios';
import { Offer } from '../models/Offer';

export async function getOffersBySearchItem(searchItem) {
  let offerResponse = await axios.post('http://localhost:3000/offers', searchItem);
  return offerResponse.data.map((offer) => new Offer(offer));
}

