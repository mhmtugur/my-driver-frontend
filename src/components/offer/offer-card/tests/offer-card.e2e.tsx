import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
import { Offer } from "../../../../models/Offer";

describe('offer card', () => {
  it('should render offer card', async () => {
    let page: E2EPage = await newE2EPage({ html: '<offer-card></offer-card>' });
    let element: E2EElement = await page.find('offer-card');
    const sampleOffer = new Offer({
      web_img: 'car.png',
      amount: 200,
      title: 'Mercedes E200',
      exampleCar: 'Example Car',
      currency: 'EUR',
    });
    element.setProperty('offer', sampleOffer);
    await page.waitForChanges();
    expect(element.innerHTML).toMatchSnapshot();
  })
});