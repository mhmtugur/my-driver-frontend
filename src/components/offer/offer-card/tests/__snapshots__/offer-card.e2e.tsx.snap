// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offer card should render offer card 1`] = `"<div class=\\"card\\"><div class=\\"card-image\\"><img src=\\"car.png\\"></div><div class=\\"card-content\\"><p class=\\"card-right-text\\">200 EUR</p><br><br><p class=\\"card-title truncate\\">Mercedes E200</p><p class=\\"truncate\\">Example Car</p></div><div class=\\"card-action\\"><a class=\\"btn waves-effect waves-light\\">SELECT</a></div></div>"`;
